docs/advanced.rst

```diff
@@ -33,26 +33,39 @@ that need it, like the ``ItemWebPage``.
         html: str
 
 
-    @provides(ResponseData)
     class ResponseDataProvider(PageObjectInputProvider):
         """This class provides ``web_poet.page_inputs.ResponseData`` instances."""
+        provided_classes = {ResponseData}
 
-        def __init__(self, response: Response):
-            """This class receives a Scrapy ``Response`` as a dependency."""
-            self.response = response
+        def __call__(self, to_provide: Set[Callable], response: Response):
+            """Builds a ``ResponseData`` instance using a Scrapy ``Response``"""
+            return [ResponseData(url=response.url,html=response.text)]
```

Comment:
```suggestion
            """Build a ``ResponseData`` instance using a Scrapy ``Response``"""
            return [ResponseData(url=response.url, html=response.text)]
```

Reply:
Always forgetting the imperative :upside_down_face: